src/tox/tox_env/python/api.py

```diff
@@ -168,9 +168,14 @@ def _validate_base_python(
         if env_base_python is not None:
             spec_name = PythonSpec.from_string_spec(env_base_python)
             for base_python in base_pythons:
-                if Path(base_python).is_absolute():
-                    return [base_python]
                 spec_base = PythonSpec.from_string_spec(base_python)
+                if spec_base.path is not None:
+                    path = Path(spec_base.path).absolute()
+                    if str(spec_base.path) == sys.executable:
+                        ver, is_64 = sys.version_info, sys.maxsize != 2**32
+                        spec_base = PythonSpec.from_string_spec(f"{sys.implementation}{ver.major}{ver.minor}-{is_64}")
```

Comment:
Moreover, @frenzymadness pointed out that the string ends with a boolean, like `...-313-True`, while according to a PATTERN in virtualenv:

https://github.com/pypa/virtualenv/blob/95c5eed96c225faff01ac2f108368e0c4303989b/src/virtualenv/discovery/py_spec.py#L8

```python
PATTERN = re.compile(r"^(?P<impl>[a-zA-Z]+)?(?P<version>[0-9.]+)?(?:-(?P<arch>32|64))?$")
```

It should probably end with `-32` or `-64` if we care about the 32/64bitness.

Moreover, when testing this assumption, I found out that the entire string is probably completely bogus:

```pycon
>>> ver, is_64 = sys.version_info, sys.maxsize != 2**32
>>> f"{sys.implementation}{ver.major}{ver.minor}-{is_64}"
"namespace(name='cpython', cache_tag='cpython-313', version=sys.version_info(major=3, minor=13, micro=0, releaselevel='candidate', serial=1), hexversion=51183809, _multiarch='x86_64-linux-gnu')313-True"
```

We probably want to replace `sys.implementation` with `sys.implementation.name` if `cpython313-64` is what you intended.